bfvmm/src/memory_manager/src/page_table_entry_x64.cpp

```diff
@@ -94,14 +94,22 @@ page_table_entry_x64::set_ps(bool enabled) noexcept
 { *m_pte = enabled ? set_bit(*m_pte, 7) : clear_bit(*m_pte, 7); }
 
 bool
-page_table_entry_x64::pat() const noexcept
+page_table_entry_x64::pat_4k() const noexcept
 { return is_bit_set(*m_pte, 7); }
 
 void
-page_table_entry_x64::set_pat(bool enabled) noexcept
+page_table_entry_x64::set_pat_4k(bool enabled) noexcept
 { *m_pte = enabled ? set_bit(*m_pte, 7) : clear_bit(*m_pte, 7); }
 
 bool
+page_table_entry_x64::pat_large() const noexcept
+{ return is_bit_set(*m_pte, 12); }
+
+void
+page_table_entry_x64::set_pat_large(bool enabled) noexcept
+{ *m_pte = enabled ? set_bit(*m_pte, 12) : clear_bit(*m_pte, 12); }
```

Comment:
Lol. Brendan, when I saw your "hell has frozen over" email, I was expecting to see functors, but this is almost as good.